src/core/lib/security/security_connector/local/local_security_connector.cc

```diff
@@ -103,12 +110,31 @@ void local_check_peer(grpc_security_connector* /*sc*/, tsi_peer /*peer*/,
     grpc_core::ExecCtx::Run(DEBUG_LOCATION, on_peer_checked, error);
     return;
   }
+  // Add TSI_SECURITY_LEVEL_PEER_PROPERTY type peer property.
+  size_t new_property_count = peer.property_count + 1;
+  tsi_peer_property* new_properties = static_cast<tsi_peer_property*>(
+      gpr_zalloc(sizeof(*new_properties) * new_property_count));
+  for (size_t i = 0; i < peer.property_count; i++) {
+    new_properties[i] = peer.properties[i];
+  }
+  if (peer.properties != nullptr) gpr_free(peer.properties);
+  peer.properties = new_properties;
+  // TODO(yihuazhang): Set security level of local TCP to TSI_SECURITY_NONE.
+  const char* security_level =
+      type == UDS ? tsi_security_level_to_string(TSI_PRIVACY_AND_INTEGRITY)
```

Comment:
Given there is a TODO, it is better to set the following for now.

`const char* security_level = tsi_security_level_to_string(TSI_PRIVACY_AND_INTEGRITY);`

Otherwise, the code does not make sense.